cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/engine/sql/dataset/SQLDataset.java

```diff
@@ -26,4 +29,11 @@ public interface SQLDataset extends SQLDatasetDescription {
    */
   long getNumRows();
 
+  /**
+   * Returns additional metrics that should be logged for this dataset.
+   */
+  default Map<String, Long> getMetrics() {
```

Comment:
Can we have a push and not pull API? This API allows to report metrics only on job finished. While it would be pretty valuable to report metrics while job is running, even if not right now. We should make an API that allows it